src/main/webapp/app/overview/course-sidebar/course-sidebar.component.ts

```diff
@@ -106,23 +106,25 @@ export class CourseSidebarComponent implements OnInit {
     applyThreshold(threshold: number, height: number) {
         const newHiddenItems: SidebarItem[] = [];
         let newAnyItemHidden = false;
-
-        const reversedSidebarItems = [...this.sidebarItems()].reverse();
-        reversedSidebarItems.forEach((item, index) => {
-            const currentThreshold = threshold - index * this.ITEM_HEIGHT;
-            item.hidden = height <= currentThreshold;
-            if (item.hidden) {
-                newAnyItemHidden = true;
-                newHiddenItems.unshift(item);
-            }
-        });
+        if (this.sidebarItems()) {
+            const reversedSidebarItems = [...this.sidebarItems()].reverse();
+
+            reversedSidebarItems.forEach((item, index) => {
+                const currentThreshold = threshold - index * this.ITEM_HEIGHT;
+                item.hidden = height <= currentThreshold;
+                if (item.hidden) {
+                    newAnyItemHidden = true;
+                    newHiddenItems.unshift(item);
+                }
+            });
+        }
 
         this.anyItemHidden.set(newAnyItemHidden);
         this.hiddenItems.set(newHiddenItems);
     }
 
     /** Calculate threshold levels based on the number of entries in the sidebar */
     calculateThreshold(): number {
-        return this.sidebarItems().length * this.ITEM_HEIGHT + this.WINDOW_OFFSET;
+        return this.sidebarItems()?.length * this.ITEM_HEIGHT + this.WINDOW_OFFSET;
```

Comment:
tbh. I'm not sure if that's the best way to handle this. If `this.sidebarItems` is undefined, the method will return NaN. I would assume that would cause some other issues along the line. Would returning a base value make sense, if `this.sidebarItems()` indeed turns out to be undefined?